File: noop-java/src/main/java/com/exyui/android/debugbottle/components/injector/Injector.java
package com.exyui.android.debugbottle.components.injector;

import android.app.Activity;
import android.util.Log;

/**
 * Created by yuriel on 9/7/16.
 */

@SuppressWarnings("unused")
public abstract class Injector {
    private static final String TAG = "FakeInjector";
    protected final Activity activity = null;
    protected final Activity getActivity() {
        Log.d(TAG, "getActivity");
        return null;
    }
    protected final void put(Object a, Object b) {
        Log.d(TAG, "put");
    }
    protected final void quickEntry(Object a, Object b, Object c) {
        Log.d(TAG, "quickEntry");
    }
    protected final void quickEntry(Object a, Object b) {
        Log.d(TAG, "quickEntry");
    }
    protected final void excludeFromMonkey(Object a) {
        Log.d(TAG, "excludeFromMonkey");
    }

    public abstract void inject();

    public void beforeMonkeyTest() {
        Log.d(TAG, "beforeMonkeyTest");
    }
}

File: noop-kotlin/src/main/kotlin/com/exyui/android/debugbottle/components/injector/Injector.kt
@file:Suppress("UNUSED_PARAMETER", "unused")

package com.exyui.android.debugbottle.components.injector

import android.app.Activity
import android.util.Log

/**
 * Created by yuriel on 8/30/16.
 */
abstract class Injector {
    private val TAG = "FakeInjector"
    protected val activity: Activity? = null
    protected fun put(a: Any? = null, b: Any? = null) {
        Log.d(TAG, "add")
    }

    protected fun quickEntry(a: Any? = null, b: Any? = null, c: Any? = null) {
        Log.d(TAG, "quickEntry")
    }

    protected fun excludeFromMonkey(a: Any? = null) {
        Log.d(TAG, "excludeFromMonkey")
    }

    abstract fun inject()
}